shared/js/abp-preprocessed.es6.js

```diff
@@ -140,3 +143,40 @@ function getVersionParam () {
 
     return versionParam
 }
+
+function fetchSurrogateCode () {
+    let trackersSurrogateListEtag = settings.getSetting('trackersSurrogateList-etag') || ''
+	// reset etag to get a new list copy if we don't have data
+	if (!trackersSurrogateList || !trackersSurrogateListEtag) trackersSurrogateListEtag = ''
+	// load surrogates list
+	load.loadExtensionFile({url: constants.trackersSurrogateList, etag: trackersSurrogateListEtag, source: 'external'}, (listData, response) => {
+	    const newTrackersSurrogateListEtag = response.getResponseHeader('etag') || ''
+	    const clientChecksum = crypto.createHash('md5').update(listData).digest('base64')
+
+	    settings.updateSetting('trackersSurrogateList-etag', newTrackersSurrogateListEtag);
+
+	    //  check that etag hash matches hash of file received
+	    if (!clientChecksum || clientChecksum.substring(0,6) !== newTrackersSurrogateListEtag) {
```

Comment:
Why is there this extra checksum step?

Reply:
We're checking that the contents of the file the extension receives match the contents of the file sent from our servers. Doing this to somewhat mitigate a potential MitM attack.